lib/pbench/server/filetree.py

```diff
@@ -340,9 +354,27 @@ def unpack(self, incoming: Path, results: Path):
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        self.unpacked = unpacked
+        command = "find unpacked/ \( -type d -exec chmod ugo+rx '{}' \; \) -o \( -type f -exec chmod ugo+r '{}' \; \)"
```

Comment:
Use `\+` instead of `\;` ... it's a lot more efficient!

However, there's no `unpacked/` directory in the current working directory, is there? (And if there is, it's probably not what we want.) I think this needs to be an f-string with `{unpacked}`. (And if you make the whole command into an f-string, you'll need to "quote" the `{}` `-exec` markers as `{{}}` so they don't get substituted.)

Also, I don't think you need `shell=True` for this, and then you probably wouldn't need the `\` quotes for the `+`, `(`, and `)` characters. (Which, come to think of it, are probably eaten by Python here in forming the string, so your shell subprocess is probably misinterpreting the parenthesized expressions as sub-shells...)

Reply:
if we make current working durectory to be incoming then we will have unpacked directory there.